dataclients/kubernetes/crd.md

```diff
@@ -0,0 +1,596 @@
+# RouteGroup Proposal
+
+Date: 2019-09-04
+Status: open
+Owners: @aryszka @szuecs
+
+This proposal will be kept up to date, if you like to discuss it
+please use https://github.com/zalando/skipper/issues/660
+
+Disclaimer:
+Skipper's `kubernetes` dataclient will also support Ingress in the
+future.
+
+## Purpose
+
+The purpose of the document is to define a CRD, that solves issues
+we observed, while using Ingress.
+
+There are some other solutions out there, like Istio, SMI and others,
+but all of these have drawbacks for skipper provided use cases. One
+common example is an application that needs a redirect to somewhere
+else for a specific path in an Ingress type of object. This example is
+only provided with multiple objects or annotations, both are unsafe
+for concurrent use.
```

Comment:
I think we need to expand on this section and show more clearly why some other existing solutions doesn't provide what we need or make it overly complex. The statement: "both are unsafe for concurrent use" is hard to validate as a reader without some small example.

I think it's important to focus on the reason why we want a single CRD which the other solutions don't provide and what we believe the benefit is of that, because feature wise we probably could implement most of it with SMI (I'm not sure) and if we can't implement all the features then this should also be highlighted as it makes it more reasonable to go with a custom solution.

Reply:
SMI does not provide access to predicates and filters, but on the other hand yes you are right I will add this.

Reply:
I changed purpose and problem statement to focus on what we want to achieve, not on weaknesses of other solutions.